lib/contributions/index.js

```diff
@@ -0,0 +1,170 @@
+    Calendar(options);
+
+    // generate the repository path
+    var repoId = Math.random().toString(36).substring(3)
+      , publicDir = Path.resolve(__dirname + "/../public/repos")
+      , repoPath = publicDir + "/" + repoId
+      , numberOfCommits = options.coordinates.length * options.commitsPerDay
+      , readmeFilePath = repoPath + "/README.md"
+      , completedCommits = 0
+      , repo = new Repo(repoPath)
+      ;
+
+    /**
+     * dummyCommit
+     * Creates a dummy commit.
+     *
+     * @name dummyCommit
+     * @function
+     * @param {String} message The commit message.
+     * @param {Function} callback The callback function.
+     * @return {undefined}
+     */
+    function dummyCommit(message, date, callback) {
+        var options = date ? "--date " + date : "";
+        Fs.writeFile(readmeFilePath, Math.random().toString(), function (err) {
+            if (err) { return callback(err); }
+            repo.add(function (err) {
+                if (err) { return callback(err); }
+                repo.commit(message, options, callback);
+            });
+        });
+    }
+
+    /**
+     * initialCommit
+     * Creates the repository and the initial commit.
+     *
+     * @name initialCommit
+     * @function
+     * @param {Function} callback The callback function.
+     * @return {undefined}
+     */
+    function initialCommit(callback) {
+        repo.create(function (err) {
+            if (err) { return callback(err); }
+            dummyCommit("Initial commit", "", callback);
+        });
+    }
+
+    /*!
+     * lastCommit
+     * Creates the last commit.
+     *
+     * @name lastCommit
+     * @function
+     * @return {undefined}
+     */
+    function lastCommit() {
+        Fs.writeFile(readmeFilePath,
+            "My GitHub Contributions Calendar"
+          + "================================"
+          + "This repository was generated with [GitHub Contributions](https://github.com/IonicaBizau/github-contributions) generator. Thanks, [@IonicaBizau](https://github.com/IonicaBizau)."
+        , function (err) {
+            if (err) { return done(err); }
+            repo.add(function (err) {
+                if (err) { return done(err); }
+                repo.commit("Added the title and description (/cc. @IonicaBizau)", function (err) {
+                    if (err) { return done(err); }
+                    var output = Fs.createWriteStream(repoPath + ".zip")
+                      , archive = Archiver("zip")
+                      , zipErr = null
+                      ;
+
+                    output.on("close", function () {
+                        done(zipErr, repoId);
+                    });
+
+                    archive.on("error", function(err){
+                        zipErr = err;
+                    });
+
+                    archive.pipe(output);
+                    archive.bulk([
+                        { expand: true, src: ["*", ".git/**/*"], cwd: repoPath, dest: repoId }
```

Comment:
@ComFreek This archives the `.git` files too. Thanks @ctalkington!

The full conversation is here: https://github.com/ctalkington/node-archiver/issues/110